app/experimenter/experiments/models/nimbus.py

```diff
@@ -257,6 +262,12 @@ def targeting(self):
                         version=self.firefox_min_version
                     )
                 )
+            if self.firefox_max_version:
+                expressions.append(
+                    "version|versionCompare('{version}') <= 0".format(
```

Comment:
Ah heck this isn't exactly right.  Looking at Normandy for reference, it looks like this is the pattern:

```
((version|versionCompare(\"70.!\") >=0 ) && (version|versionCompare(\"70.*\") < 0))
```

so we'll need to replace the `!` with a `*` and compare `< 0`

So we could either leave the constants as they are with `!` for the min case, explicitly replace it with a `*` here in the max version clause, and call that good enough.  

Or we could take `!` out of the min case and then explicitly add it in the min case and add `*` in the max case.  Of course we have to remember to support cases where neither `!` and `*` are used when a specific version is hard coded.  But this seems like a hairer solution so maybe the first option is better.

Reply:
Found the normandy implementation:

https://github.com/mozilla/normandy/blob/master/normandy/recipes/filters.py#L634-L635

Reply:
Yeah, I'll just go with a string replace of "!" with "*" for now.